reporter/src/main/kotlin/reporters/NoticeTemplateReporter.kt

```diff
@@ -19,15 +19,202 @@
+        }
+
+        return outputFiles
+    }
+
+    /**
+     * License information for a single package or project.
+     */
+    class PackageNoticeModel(
+        val id: Identifier,
+        private val input: ReporterInput
+    ) {
+        /**
+         * True if the package is excluded.
+         */
+        val excluded: Boolean by lazy { input.ortResult.isExcluded(id) }
+
+        /**
+         * The resolved license information for the package.
+         */
+        val license: ResolvedLicenseInfo by lazy {
+            val resolved = input.licenseInfoResolver.resolveLicenseInfo(id)
+            resolved.copy(licenses = resolved.licenses.sortedBy { it.license.toString() })
+        }
+
+        /**
+         * The resolved license file information for the package.
+         */
+        val licenseFiles: ResolvedLicenseFileInfo by lazy { input.licenseInfoResolver.resolveLicenseFiles(id) }
+
+        /**
+         * Returns all [ResolvedLicense]s for this package excluding those licenses which are contained in any of the
+         * license files. This is useful when the raw texts of the license files are included in the generated notice
+         * file and all licenses not contained in those files shall be listed separately.
+         */
+        fun licensesNotInLicenseFiles(): List<ResolvedLicense> {
+            val noticeFileLicenses = licenseFiles.files.flatMap { it.licenses }
+            return license.filter { it !in noticeFileLicenses }
+        }
+    }
+
+    /**
+     * A collection of helper functions for the Freemarker templates.
+     */
+    class NoticeHelper(private val licenseConfiguration: LicenseConfiguration) {
+        /**
+         * Filter licenses that are configured to be [included in the notice file][License.includeInNoticeFile] in the
+         * [LicenseConfiguration].
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun filterIncludeInNoticeFile(licenses: List<ResolvedLicense>): List<ResolvedLicense> =
+            licenses.filter { resolvedLicense ->
+                licenseConfiguration.licenses.find { it.id == resolvedLicense.license }?.includeInNoticeFile ?: true
+            }
+
+        /**
+         * Return a [LicenseView] constant by name to make them easily available to the Freemarker templates.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun licenseView(name: String): LicenseView =
+            LicenseView.Companion::class.memberProperties
+                .first { it.name == name }
+                .get(LicenseView.Companion) as LicenseView
+
+        /**
+         * Merge the [ResolvedLicense]s of multiple [models] and filter them using [licenseView]. [Omits][omitExcluded]
+         * excluded packages, licenses, and copyrights by default. The returned list is sorted by license identifier.
+         */
+        @JvmOverloads
+        fun mergeLicenses(
```

Comment:
IDE inspections seems to thing that this also needs

        @Suppress("UNUSED") // This function is used in the templates.

Reply:
Weird, it didn't highlight the function for me. Done.